src/Featurizers/Components/FilterFeaturizerImpl.h

```diff
@@ -0,0 +1,641 @@
+
+    using BaseType =
+        Transformer<
+            InputTupleT,
+            typename TheseFilterTransformerTraits::TransformedType
+        >;
+
+    using ChainedTransformerUniquePtr =
+        std::unique_ptr<
+            Transformer<
+                typename ChainedTransformerT::InputType,
+                typename ChainedTransformerT::TransformedType
+            >
+        >;
+
+    // ----------------------------------------------------------------------
+    // |
+    // |  Public Methods
+    // |
+    // ----------------------------------------------------------------------
+    template <typename... ChainedTransformerConstructArgTs>
+    FilterTransformer(ChainedTransformerConstructArgTs &&... args);
+    FilterTransformer(ChainedTransformerUniquePtr pTransformer);
+    FilterTransformer(Archive &ar);
+
+    ~FilterTransformer(void) override = default;
+
+    FEATURIZER_MOVE_CONSTRUCTOR_ONLY(FilterTransformer);
+
+    bool operator==(FilterTransformer const &other) const;
+    bool operator!=(FilterTransformer const &other) const;
+
+    void save(Archive &ar) const override;
+
+private:
+    // ----------------------------------------------------------------------
+    // |
+    // |  Private Data
+    // |
+    // ----------------------------------------------------------------------
+    ChainedTransformerUniquePtr             _pTransformer;
+
+    // ----------------------------------------------------------------------
+    // |
+    // |  Private Methods
+    // |
+    // ----------------------------------------------------------------------
+    void execute_impl(typename BaseType::InputType const &input, typename BaseType::CallbackFunction const &callback) override {
+        _pTransformer->execute(
+            TheseFilterTransformerTraits::ToChainedInputType(input),
+            [&input, &callback](typename ChainedTransformerT::TransformedType value) {
+                callback(TheseFilterTransformerTraits::ToTransformedType(input, std::move(value)));
+            }
+        );
+    }
+
+    void flush_impl(typename BaseType::CallbackFunction const &/*callback*/) override {
+        _pTransformer->flush(
+            [/*&callback*/](typename ChainedTransformerT::TransformedType /*value*/) {
+                // If errors are raised here, it is because the ChainedTransformer has dynamic output.
+                // This functionality isn't implemented right now, as we aren't sure of a good
+                // way to augment `value` with the row information provided when the corresponding
+                // call to `execute` was made. Furthermore, we don't have a good way to associated
+                // calls to this callback with multiple calls to `execute`. Further design changes
+                // will be necessary if this functionality is required in the future.
+                throw std::runtime_error("Transformers that rely on flush can't be wrapped in this object at this time");

```

Comment:
I think they may be an issue earlier than I was thinking. When we first talked about this I was thinking we would only have to wrap 1 step of the pipeline. After our conversations I realize that if a "normal" featurizer that is wrapped in the filter featurizer returns output, then the rest of the featurizers in the pipeline will need to be filtered as well. This means that if an earlier step is not dynamic, but a final step is, that we need to include this functionality.

Fortunately, in this current case the featurizer that is getting wrapped only produces an annotation and not any output, so it is still not an issue. I just wanted to call it out.